SixTrack/cmake_six

```diff
@@ -0,0 +1,255 @@
+	echo "clean : Remove all build directories"
+	echo "help  : Display this message"
+
+	echo ""
+	echo "Examples:"
+	echo "cmake_six gfortran COLLIMAT -CRLIBM"
+	echo "cmake_six clean"
+	echo "cmake_six ifort-icc"
+	echo "cmake_six BIGNPART STF"
+	exit
+    fi
+done
+
+options_ALL=() # All set options
+options_ON=()  # All options switched ON
+options_OFF=() # All options switched OFF
+
+for var in "$@" #Loop over command line arguments
+do
+    if [[ $var == -* ]]; then ## Negative options
+	#echo "OFF" $var
+	varOn=${var:1} # Without the "-"
+	for v in ${options_ALL[@]}
+	do
+	    if [[ $v == $varOn ]]; then
+		echo "ERROR: option $v used more than once"
+		exit 1
+	    fi
+	done
+
+	for v in ${compilers[@]}
+	do
+	    if [[ $v == $varOn ]]; then
+		echo "ERROR: can't switch a compiler off"
+		exit 1
+	    fi
+	done
+	
+	options_ALL+=(${var:1})
+	options_OFF+=(${var:1})
+    else                      ## Positive options
+	#echo "ON" $var
+	varOn=$var
+	for v in ${options_ALL[@]}
+	do
+	    if [[ $v == $varOn ]]; then
+		echo "ERROR: option $v used more than once"
+		exit 1
+	    fi
+	done
+
+	thisWasCompiler="NO"
+	for v in ${compilers[@]}
+	do
+	    if [[ $v == $varOn ]]; then
+		if [[ $options_COMPILER != "defaultcompiler" ]]; then
+		    echo "ERROR: can't set multiple compilers"
+		    exit 1
+		fi
+		options_COMPILER=$varOn
+		thisWasCompiler="YES"				    
+	    fi
+	done
+
+	if [[ $thisWasCompiler == "NO" ]]; then
+	    options_ALL+=($var)
+	    options_ON+=($var)	
+	fi
+	thisWasCompiler="NO"
+    fi
+done
+#Sort the options
+# http://stackoverflow.com/questions/7442417/how-to-sort-an-array-in-bash
+IFS=$'\n' options_sorted_ALL=($(sort <<<"${options_ALL[*]}"))
+IFS=$'\n' options_sorted_ON=($(sort <<<"${options_ON[*]}"))
+IFS=$'\n' options_sorted_OFF=($(sort <<<"${options_OFF[*]}"))
+
+echo "options_ON  =" ${options_sorted_ON[@]}
+echo "options_OFF =" ${options_sorted_OFF[@]}
+
+#Build compiler options
+CMAKE_OPTIONS_LIST=""
+for i in $(seq 0 $(expr ${#compilers[@]} - 1 ) )
+do
+    #echo $i ${compilers[$i]} ${compilers_CMAKE[$i]}
+    if [ $options_COMPILER == ${compilers[$i]} ]; then
+	#echo "FOUND!"
+	CMAKE_OPTIONS_LIST+=${compilers_CMAKE[$i]}
+    fi
+done
+
+#Make folder & build cmake options list
+NAME="SixTrack_cmakesix"
+
+for opt in ${options_sorted_ON[@]}
+do
+    NAME+=_$opt
+    CMAKE_OPTIONS_LIST+=" -D$opt=ON"
+done
+for opt in ${options_sorted_OFF[@]}
+do
+    NAME+=_NO$opt
+    CMAKE_OPTIONS_LIST+=" -D$opt=OFF"
+done
+NAME+=_$options_COMPILER
+
+echo "NAME= '"$NAME"'"
+
+echo "CMAKE options = '"$CMAKE_OPTIONS_LIST"'"
+echo ""
+
+
+#Run cmake
+if [ -d $NAME ]; then
+    echo "Folder '"$NAME"' already exists, deleting!"
+    rm -rf $NAME
+fi
+mkdir $NAME
+cd $NAME
+
+set +e #From here on we catch errors manually...
+
+#Use EVAL to split the options from $CMAKE_OPTIONS_LIST
+# http://stackoverflow.com/questions/6087494/bash-inserting-quotes-into-string-before-execution
+eval cmake  $CMAKE_OPTIONS_LIST ../../.
```

Comment:
It would be great of cmake would fail (stop and return non-zero status) in case of unknown options. I've tried to make this shell script as lightweight as possible, so any real option checking is done at the cmake level.